integrations/sonarqube/client.py

```diff
@@ -468,6 +472,39 @@ async def get_all_sonarqube_analyses(
             )
             yield analysis_data
 
+    async def _get_all_portfolios(self) -> list[dict[str, Any]]:
+        logger.info(f"Fetching all portfolios in organization: {self.organization_id}")
+        response = await self.send_api_request(endpoint=Endpoints.PORTFOLIOS)
+        return response.get("views", [])
+
+    async def get_portfolio_details(self, portfolio_key: str) -> dict[str, Any]:
+        logger.info(f"Fetching portfolio details for: {portfolio_key}")
+        response = await self.send_api_request(
+            endpoint=Endpoints.PORTFOLIO_DETAILS,
+            query_params={"key": portfolio_key},
+        )
+        return response
+
+    def _get_portfolio_subportfolios(
```

Comment:
we can rename the function to _fetch_subportfolios or _extract_subportfolios to make it less verbose